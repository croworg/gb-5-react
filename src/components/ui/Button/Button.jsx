import React from 'react';
import MuiButton from "@mui/material/Button";

export const Button = (props) => {
    return (
        <>
            <MuiButton
                style={{color: 'white'}}
                color={'success'}
                size={'small'}
                variant={'contained'}
                {...props}
            >{props.children}</MuiButton>
        </>
    );
};
