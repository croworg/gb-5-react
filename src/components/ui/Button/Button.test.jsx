import React from 'react';
import {render, screen} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import '@testing-library/jest-dom';

import {Button} from "./Button";

describe('testing Button', () => {
    it('render component button', () => {
        render(<Button>Change name</Button>);
    })

    it('render with snapshot', () => {
        const {asFragment} = render(<Button>Change name</Button>);
        expect(asFragment()).toMatchSnapshot();
    })

    it('render component with text name', () => {
        render(<Button>Change name</Button>);
        expect(screen.getByText(/Change name/)).toBeInTheDocument();
    })

    it('render multiply component', () => {
        render(
            <>
                <Button>Edit</Button>
                <Button>Delete</Button>
            </>
        )
        expect(screen.queryAllByRole('button').length).toBe(2);
    })

    it('button is disabled', () => {
        render(<Button disabled>Delete</Button>);
        expect(screen.getByText(/Delete/)).toBeDisabled();
    })

    it('button have specific style', () => {
        render(<Button>Delete</Button>);
        expect(screen.getByText(/Delete/)).toHaveStyle({color: 'white'});
    })

    it('click with user event', async () => {
        const mockHandler = jest.fn();

        render(<Button onClick={mockHandler}>Delete</Button>);
        await userEvent.click(screen.getByText(/Delete/));
        expect(mockHandler).toHaveBeenCalledTimes(1);
    })

    it('test example', async () => {
        const onChange = jest.fn();
        render(<input type={"checkbox"} onChange={onChange} />);

        const checkBox = screen.getByRole('checkbox');
        await userEvent.dblClick(checkBox);
        expect(onChange).toHaveBeenCalledTimes(2);
        expect(checkBox).not.toBeChecked();
    })
})