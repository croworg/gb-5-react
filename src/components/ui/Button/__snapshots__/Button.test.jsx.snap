// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Button render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeSmall MuiButton-containedSizeSmall css-1ph36rw-MuiButtonBase-root-MuiButton-root"
    style="color: white;"
    tabindex="0"
    type="button"
  >
    Change name
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
